### PR TITLE
Md 78 implements company crud

### PR DESCRIPTION
**Pull request by claudiocassimiro**

### CODE REVIEW

File: src/app/utils/createSlug.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const createSlug = (text: string) => {
+  let result = text.toLowerCase();
+
+  result = result.replace(/\s/g, `-`);
+  result = result.normalize(`NFD`).replace(/[\u0300-\u036f]/g, ``);
+  result = result.replace(/[^a-z0-9-]/g, ``);
+  result = result.replace(/-+/g, `-`);
+
+  return result;
+};
```
Comment (EricksonSiqueira) [inline src/app/utils/createSlug.ts:10]:
```suggestion
export const createSlug = (text: string) => text.toLowerCase().replace(/\s/g, `-`).normalize(`NFD`).replace(/[\u0300-\u036f]/g, ``).replace(/[^a-z0-9-]/g, ``).replace(/-+/g, `-`);
```
I think the function name is very clear so we can use this one liner and still be readable.

File: src/middleware.ts [hunk 2]
```diff
@@ -16,6 +16,15 @@ export default authMiddleware({
...
   publicRoutes: [`/`],
+
+  // publicRoutes: [ // to test all routes on postman without need to sign in
+  //   `/`,
+  //   `/api/company/createCompany`,
+  //   `/api/company/getCompanies`,
+  //   `/api/company/getCompanyById`,
+  //   `/api/company/editCompany`,
+  //   `/api/company/deleteCompany`,
+  // ],
```
Comment (EricksonSiqueira) [inline src/middleware.ts:27]:
```suggestion
  publicRoutes: process.env.NODE_ENV !== `production` ? [
    `/`,
    `/api/company/createCompany`, 
    `/api/company/getCompanies`, 
    `/api/company/getCompanyById`, 
    `/api/company/editCompany`, 
    `/api/company/deleteCompany`
  ] : [`/`],
```
Does this work?

File: package.json [hunk 3]
```diff
@@ -30,7 +30,8 @@
     "react-dom": "18.2.0",
     "tailwindcss": "3.3.2",
     "typescript": "5.1.3",
-    "vite-tsconfig-paths": "^4.2.0"
+    "vite-tsconfig-paths": "^4.2.0",
+    "yup": "^1.2.0"
```
[the remark this answers is not in the document]
Comment (claudiocassimiro) [reply package.json:34]: make sense

File: src/app/utils/createSlug.ts (hunk 1 above)
Comment (claudiocassimiro) [reply src/app/utils/createSlug.ts:10]: Good, done

**Review by EricksonSiqueira — APPROVED** [review]

**Pull request merged**